app/src/main/java/com/example/zingmp3phake/UI/fragment/ExploreFragment.kt

```diff
@@ -0,0 +1,90 @@
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import com.example.zingmp3phake.UI.adapter.ListenerItemClickExplore
+import com.example.zingmp3phake.UI.adapter.RecyclerViewAdapter
+import com.example.zingmp3phake.data.model.Song
+import com.example.zingmp3phake.data.repo.SongRepo
+import com.example.zingmp3phake.data.repo.resource.local.LocalSong
+import com.example.zingmp3phake.data.repo.resource.remote.RemoteSong
+import com.example.zingmp3phake.databinding.FragmentExploreBinding
+import com.example.zingmp3phake.presenter.ExploreFragmentPresenter
+import com.example.zingmp3phake.presenter.IExploreFragment
+import com.example.zingmp3phake.utils.NO_INTERNET
+import com.example.zingmp3phake.utils.base.BaseFragment
+
+class ExploreFragment :
+    BaseFragment(),
+    IExploreFragment.view,
+    RecyclerViewAdapter.ItemClickListener {
+    private lateinit var binding: FragmentExploreBinding
+    private var explorePresenter: ExploreFragmentPresenter? = null
+    private val adapterRv = RecyclerViewAdapter(this)
+    private lateinit var mContext: Context
+    private var listener: ListenerItemClickExplore? = null
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        if (container != null) {
+            explorePresenter = ExploreFragmentPresenter(
+                SongRepo.getInstance(
+                    LocalSong.getInstance(container.context.applicationContext),
+                    RemoteSong.getInstance(container.context.applicationContext),
+                ),
+                this
+            )
+        }
+        if (container != null) {
+            mContext = container.context
+        }
+        binding = FragmentExploreBinding.inflate(layoutInflater)
+        return binding.root
+    }
+
+    override fun initView() {
+        binding.recycleView.adapter = adapterRv
+    }
+
+    override fun initData() {
+        if (explorePresenter?.isConnectedInternet(mContext.applicationContext) == true) {
+            explorePresenter?.getTrendingSong()
+        } else {
+            Toast.makeText(mContext.applicationContext, NO_INTERNET, Toast.LENGTH_LONG).show()
+        }
+    }
+
+    override fun displaySuccess(list: MutableList<Song>) {
+        adapterRv.setData(list)
+    }
+
+    override fun displayFail(msg: String) {
+        // TODO later
+    }
+
+    override fun onItemClick(pos: Int, listSong: MutableList<Song>) {
+        explorePresenter?.handlerStartSong(listSong, pos, mContext.applicationContext)
+        listener?.notification()
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        explorePresenter?.stopService()
+    }
+
+    fun setListenerItemClickExplore(listener: ListenerItemClickExplore) {
+        this.listener = listener
+    }
+
+    companion object {
+        fun getInstance() = synchronized(this) {
```

Comment:
```suggestion
        fun newInstance() = synchronized(this) {
```
tương tự fragment khác